sharpy/generators/polaraeroforces.py

```diff
@@ -213,10 +216,197 @@ def generate(self, **params):
+
+        The total forces and moments are then computed as a summation of the above, where the total forces
+
+        are computed simply as
+
+        .. math:: K_F \leftarrow \Sum w_f^i
+
+        and the total moments
+
+        .. math:: K_M \leftarrow \Sum \tilde{R}_{A,i} w_f^i + \Sum w_m^i C^{AB} T^{-\top}
+
+        which leaves the end result expressed in the A frame.
+
+        An additional gain, referred to as ``forces_at_ref_gain`` is needed to obtain the nodal force at the
+        :math:`A` frame in order to correct the forces at this point with the local polar. Thus, this gain
+        subtracts from the total forces and moments the sum of those from the remaining nodes. Then the ``polar_gain``
+        corrects the local force at the A frame and sums the corrected forces and moments at the other nodes.
+
+        Keyword Args:
+            beam (sharpy.linear.assembler.linearbeam.LinearBeam): Beam object
+            tsstruct0 (sharpy.utils.datastructures.StructTimestepInfo): Ref structural time step
+            tsaero0 (sharpy.utils.datastructures.AeroTimestepInfo): Ref aero time step info
+
+        Returns:
+            np.array: Gain matrix
+        """
+        beam = params['beam']
+        tsstruct0 = params['tsstruct0']
+        tsaero0 = params['tsaero0']
+
+        aerogrid = self.aero
+        structure = self.structure
+
+        structural_kstep = tsstruct0
+        aero_kstep = tsaero0
+        nnode = tsstruct0.pos.shape[0]
+
+        aero_dict = aerogrid.aero_dict
+        if aerogrid.polars is None:
+            return None
+
+        cga = algebra.quat2rotation(structural_kstep.quat)
+
+        # Matrix allocation
+        num_dof_str = beam.sys.num_dof_str
+        num_dof_rig = beam.sys.num_dof_rig
+        use_euler = beam.sys.use_euler
+
+        # GEBM degrees of freedom
+        jj_for_tra = range(num_dof_str - num_dof_rig,
+                           num_dof_str - num_dof_rig + 3)
+        jj_for_rot = range(num_dof_str - num_dof_rig + 3,
+                           num_dof_str - num_dof_rig + 6)
+
+        if use_euler:
+            jj_euler = range(num_dof_str - 3, num_dof_str)
+            euler = algebra.quat2euler(tsstruct0.quat)
+            tsstruct0.euler = euler
+        else:
+            jj_quat = range(num_dof_str - 4, num_dof_str)
+
+        polar_gain = np.zeros((num_dof_str, num_dof_str))
+
+        # this gain computes the aerodynamic forces and moments at the node at the A frame, which else
+        # would be included in the total force and total moment. This is needed to correct these nodal forces too
+        forces_at_ref_gain = np.zeros_like(polar_gain)
+
+        jj = 0  # Global DOF index
+        for inode in range(nnode):
+            if aero_dict['aero_node'][inode]:
+
+                ### detect bc at node (and no. of dofs)
+                bc_here = structure.boundary_conditions[inode]
+
+                if bc_here == 1:  # clamp (only rigid-body)
```

Comment:
On a trim analysis, this very first node with bc=1 would still have DoF, therefore is this routine only valid  for A-frame node clamped conditions? It is fine if that's the case but we should remark it at the function docstring.

Reply:
The issue with the bc=1 node is that it doesn't have flexible DOF and hence the airforces at that node are not mapped to it but rather to the A frame (i.e. with the sum of all the forces). Therefore, to correct the aero force at the bc=1 node, I subtract all the other forces from the A frame, correct them, and sum them all up again:
https://github.com/ImperialCollegeLondon/sharpy/blob/684e5c30ff22a5068f867a255504c85dce7ba0b2/sharpy/generators/polaraeroforces.py#L385